packages/connection-storage/src/connection-storage.ts

```diff
@@ -102,7 +112,7 @@ export class ConnectionStorage {
     }
   }
 
-  private static async getConnections(): Promise<any[]> {
+  private static async getConnections(): Promise<ConnectionWithLegacyProps[]> {
```

Comment:
Why do we need those props to get to this layer? here they shouldn't be needed. Can we just make sure that we save them and we don't read them back?

Reply:
Ah wait that's what we are doing already?

Reply:
Nope .. I tried, but i'm a still confused :), why do we read them back? And how is `isFavorite` and `name` related to this change?

If we want to be aware of those, shouldn't it be to remove them from what we read so it never reaches other layers?

And on save we can just re-add them back.

Reply:
this method (`getConnections`), just reads all the data from the storage including legacy connections. i just added types to support that as its the expected output from this method. and then in `loadAll` method, we only consider connections with `connectionInfo` and other props (`isFavorite` and `name`) are used in `getLegacyConnections` which's used in connection migration modal (where we show list of legacy connections).

Reply:
Ahh got it! so `isFavorite` and `name` is not related to this specific `_id` thing! Makes sense, thanks!